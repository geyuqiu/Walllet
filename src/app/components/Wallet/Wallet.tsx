import React, {useState} from "react";

import {useFetch} from '../../hooks/useFetch';
import {Table} from '../Table/Table';
import {tableColumns} from '../Table/TableColumns';
import {Transaction} from '../TransactionRow/model';
import {TransactionRow} from '../TransactionRow/TransactionRow';

export const Wallet = () => {
	const [addressSelected, setAddressSelected] = useState<string>('D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax');
	const {addresses, transactions} = useFetch(addressSelected);

	return (
		<>
			{transactions.length
				? <Table columns={tableColumns} data={transactions}>
						{(transaction: Transaction, index: number) => (
							<TransactionRow transaction={transaction}/>
						)}
				</Table>
				: <div/>
			}
		</>
	);
};
