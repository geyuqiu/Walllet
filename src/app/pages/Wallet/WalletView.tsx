import React, {useState} from "react";

import {Table} from '../../components/Table/Table';
import {useFetch} from '../../hooks/useFetch';
import {Wallet} from './model';
import {Transaction} from './TransactionRow/model';
import {feeAccessor, tableColumns, timestampAccessor} from './TransactionRow/TableColumns';
import {TransactionRow} from './TransactionRow/TransactionRow';
import {WalletCard} from './WalletCard/WalletCard';

export interface HideColumnsResponsive {
	accessor: string;
	breakPoint: number;
}

const hideColumnsAtBreakpoint: HideColumnsResponsive[] = [
	{accessor: timestampAccessor, breakPoint: 1280},
	{accessor: feeAccessor, breakPoint: 1024}
];

export const WalletView = () => {
	const [isLoadingTransactions, setIsLoadingTransactions] = useState(true);
	const [wallet, setWallet] = useState<Wallet | null>({
		address: 'AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW',
		balance: '1004310900000'
	});
	const {wallets, transactions} = useFetch(wallet, setIsLoadingTransactions);

	const addressOnSelect = (address: string) => {
		const next = wallets.find(w => w.address ===  address)!;
		setWallet(next);
	}

	return (
		<>
			<WalletCard wallets={wallets} wallet={wallet} addressOnSelect={addressOnSelect}/>
			<section className="ml-6 mr-3 sm:mx-8 sm:flex sm:justify-center pt-12">
				{isLoadingTransactions && <p>Loading!</p>}
				{!isLoadingTransactions && !transactions.length && <p>No transactions were found for this wallet!</p>}
				{!isLoadingTransactions && transactions?.length > 0 &&
					<Table columns={tableColumns} data={transactions} hideColumnsAtBreakpoint={hideColumnsAtBreakpoint}>
						{(transaction: Transaction, index: number) => (
							<TransactionRow transaction={transaction} address={wallet!.address}/>
						)}
					</Table>
				}
			</section>
		</>
	);
};
