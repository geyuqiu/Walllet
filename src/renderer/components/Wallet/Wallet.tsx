import React from "react";

import {useFetch} from '../../hooks/useFetch';

export const Wallet = () => {
	const {wallets} = useFetch();

	return <div>{JSON.stringify(wallets)}</div>;
};
