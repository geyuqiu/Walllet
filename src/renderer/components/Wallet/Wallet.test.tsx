import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { Wallet } from "./Wallet";

describe("Wallet", () => {
	it("should match snapshot", () => {
		const { container } = render(
			<MemoryRouter>
				<Wallet />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
