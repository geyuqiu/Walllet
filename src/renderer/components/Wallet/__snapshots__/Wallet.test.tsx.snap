// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet should match snapshot 1`] = `
<div>
  <div>
    []
  </div>
</div>
`;
